File: src/Entities/Snake.ts
import { Entity } from './Entity';
import { SnakeController } from '../Components/SnakeController';
import { SnakeRenderer } from '../Components/SnakeRenderer';
import { Vector2 } from '../Core/Vector';
import { canvas } from '../index';

export class Snake extends Entity {
    snakeController: SnakeController;

    constructor() {
        super();
        this.pos = new Vector2(canvas.width / 2, canvas.height / 2);

        this.snakeController = new SnakeController(this);
        super.addComponent(this.snakeController);

        const sr = new SnakeRenderer(this);
        super.addComponent(sr);
    }
}

File: lukewarm-snake/Entities/Snake.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using BlackMagic;
using Microsoft.Xna.Framework;

namespace lukewarm_snake
{
    public class Snake : Entity
    {
        SnakeController snakeController;
        TailHandler tailHandler;
        SnakeRenderer tailRenderer;
        SnakeRippleHandler rippleHandler;
        FoodEater snakeCollider;
        SnakeHealth health;
        public Snake(int startSnakeLength = TailHandler.minAnchors) : base(new Vector2(Globals.Camera.Width, Globals.Camera.Height) / 2f)
        {
            health = new(this);
            AddTrait(health);

            tailHandler = new(this, startAnchors: startSnakeLength);
            AddTrait(tailHandler);

            tailRenderer = new(this, tailHandler, health);
            AddTrait(tailRenderer);

            rippleHandler = new(this);
            AddTrait(rippleHandler);

            snakeController = new(this, tailHandler);
            AddTrait(snakeController);

            snakeCollider = new(this);
            AddTrait(snakeCollider);

        }
    }
}
